test/decl/subscript/subscripting.swift

```diff
@@ -246,18 +246,75 @@ struct MutableSubscriptInGetter {
 }
 
 func testSubscript1(_ s1 : SubscriptTest1) {
   let _ : Int = s1["hello"]  // expected-error {{ambiguous subscript with base type 'SubscriptTest1' and index type 'String'}}
-  
+
   if s1["hello"] {}
-  
-  
-  let _ = s1["hello"]  // expected-error {{ambiguous use of 'subscript'}}
+
+  _ = s1.subscript((true, (5, SubClass())), "hello")
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{52-53=]}}
+  _ = s1.subscript((true, (5, baz: SubSubClass())), "hello")
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{60-61=]}}
+  _ = s1.subscript((fo: true, (5, baz: SubClass())), "hello")
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}}
+  _ = s1.subscript(SubSubClass())
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{33-34=]}}
+  _ = s1.subscript(ClassConformingToProtocol())
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{47-48=]}}
+  _ = s1.subscript(ClassConformingToRefinedProtocol())
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{54-55=]}}
+  _ = s1.subscript(true)
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}}
+  _ = s1.subscript(SuperClass())
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}}
+  _ = s1.subscript("hello")
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{27-28=]}}
+  _ = s1.subscript("hello"
+  // expected-error@-1 {{type 'SubscriptTest1' has no member property or method named 'subscript'}}
+  // expected-note@-2 {{did you mean to use the subscript operator?}} {{9-10=}} {{10-19=}} {{19-20=[}} {{27-27=]}}
+  // expected-note@-3 {{to match this opening '('}}
+
+  let _ = s1["hello"]
+  // expected-error@-1 {{ambiguous use of 'subscript'}}
+  // expected-error@-2 {{expected ')' in expression list}}
```

Comment:
I wonder what is going on here?

Reply:
The last test of mine doesn't have a closing parenthesis to make sure the fix-it behaves correctly in this case. That spawnes the `expected ')'` error as well.

Reply:
Ah I didn't know that we did that but ok :(

Reply:
IMO it's a nice addition considering the fixit only appears if the inputs are compatible, so the user either has or hasn't typed a `)` before paying attention to the error. If you have something to say, I'll listen.

Reply:
Yes, I think it's fine, I was just surprised that it happens this way :) No worries!